managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedLedgerImpl.java

```diff
@@ -2762,6 +2762,7 @@ void internalTrimLedgers(boolean isTruncate, CompletableFuture<?> promise) {
                         config.getLedgerOffloader().getOffloadDriverMetadata());
                 OffloadUtils.setOffloadDriverMetadata(newInfoBuilder, driverName, driverMetadata);
                 ledgers.put(ls.getLedgerId(), newInfoBuilder.build());
+                invalidateReadHandle(ls.getLedgerId());
```

Comment:
We need to check if the LedgerHandle is bookie LedgerHandle. If we configure the read policy to  tiered storage first, the LedgerHandle will be tiered storage Ledger Handle. Invalidate the ReadHandle when removing the Ledger from BookKeeper will lead to NPE if the current LedgerHandle is reading data from tiered storage.

Reply:
got. updated.